arroyo/backends/kafka/consumer.py

```diff
@@ -157,7 +158,9 @@ def __init__(
         if commit_retry_policy is None:
             commit_retry_policy = NoRetryPolicy()
 
+        configuration = dict(configuration)
         auto_offset_reset = configuration.get("auto.offset.reset", "largest")
+        self.__force_offset_reset = configuration.pop("force.offset.reset", False)
```

Comment:
I will admit that having this in the general kafka configuration is really ugly but it's by far the easiest one given the current way this is used in snuba.

Reply:
Are you sure it is not equally complex to pass the flag as an `__init__` parameter of the consumer (there are like 2-3 implementations), then update the `consumer_builder` which basically has access to most CLI parameters. That one creates the Consumer almost everywhere (except for subscriptions and replacements).

Reply:
@fpacifici now that I thought about this more I think the weird thing is that we're basically overriding the real `auto.offset.reset` later in some of the consumers. Maybe we actually should approach it the other way round and leave `auto.offset.reset` untouched and make it behave like `force.offset.reset` and then have a separate flag somewhere to override the behavior for what is intended in prod?

Reply:
Yes I think that would be the right way.

Reply:
I think I will do this change. In that case though we need to be mindful to not update arroyo independent of the changes in snuba or we lose the intended behavior.

Reply:
I inversed the logic now. There is now a flag called `arroyo.strict.offset.reset` which defaults (if not set) to `True` which keeps the current behavior. If the flag is set to `False` then the regular kafka logic kicks in of honoring the auto offset reset.